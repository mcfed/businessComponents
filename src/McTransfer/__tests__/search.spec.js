import React from 'react';
import {shallow} from 'enzyme';
import Search from '../ts/search.tsx';

function setup(props, options) {
  const defaultProps = {
    placeholder: '',
    value: '',
    prefixCls: '',
    searchRender: '',
    handleClear: jest.fn(),
    onChange: jest.fn()
  };
  const wrapper = shallow(
    <Search {...Object.assign({}, defaultProps, props)} />,
    options
  );

  return {
    wrapper
  };
}

describe('快照测试', () => {
  it('全页快照', () => {
    const props = {
      item: {}
    };
    const {wrapper} = setup(props);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('方法测试全覆盖', () => {
  const props = {
    item: {
      disabled: false
    }
  };
  it('handleChange 方法测试', () => {
    const {wrapper} = setup(props);
    const instance = wrapper.instance();
    instance.handleChange();
  });
});
