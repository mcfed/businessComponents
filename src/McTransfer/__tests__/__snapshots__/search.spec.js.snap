// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 全页快照 1`] = `
<div
  className="-wapper"
>
  <Component
    className=""
    onChange={[Function]}
    placeholder=""
    value=""
  />
</div>
`;
